engine/include/cubos/engine/ui/canvas/draw_list.hpp

```diff
@@ -0,0 +1,74 @@
+#include <vector>
+
+#include <cubos/core/gl/render_device.hpp>
+
+#include <cubos/engine/ui/canvas/draw_command.hpp>
+#include <cubos/engine/ui/canvas/draw_type.hpp>
+
+namespace cubos::engine
+{
+    /// @brief Describes a draw commands and how it fits into the draw list carrying it.
+    /// @ingroup ui-canvas-plugin
+    struct UIDrawListEntry
+    {
+        UIDrawType& type;
+        UIDrawCommand command;
+        size_t offset;
+    };
+
+    /// @brief Holds a collections of draw commands and their data.
+    /// @ingroup ui-canvas-plugin
+    class UIDrawList
```

Comment:
```suggestion
    class CUBOS_ENGINE_API UIDrawList
```
Also missing in other struct and class definitions.